static/js/starred_messages.js

```diff
@@ -43,6 +45,30 @@ export function get_starred_msg_ids() {
     return Array.from(starred_ids);
 }
 
+export function get_starred_msg_ids_in_topic(stream_id, topic) {
+    if (stream_id === undefined || topic === undefined) {
+        return [];
+    }
+
+    return Array.from(starred_ids).filter((id) => {
+        const message = message_store.get(id);
+
+        if (message === undefined) {
+            // We know the `id` from the initial data fetch from page_params,
+            // but the message itself hasn't been fetched from the server yet.
+            // The message with this `id` may belong to some other topic,
+            // so don't return it.
+            return false;
```

Comment:
I think we should mention here that this means that our implementation is not correct, and that to fix this we will need to ask the server for the set of message IDs to modify.

(I think it's OK to merge this feature with that known bug and address that as a follow-up, to manage complexity and because I expect this bug to be not easy to trigger).

Reply:
Updated the comment and commit message to mention this. I think it was decided that the feature would be frontend only in V1? https://chat.zulip.org/#narrow/stream/137-feedback/topic/unstar.20in.20topic/near/735823